src/wallet/wallet.cpp

```diff
@@ -2220,7 +2220,14 @@ CAmount CWallet::GetAvailableBalance(const CCoinControl* coinControl) const
 
     CAmount balance = 0;
     std::vector<COutput> vCoins;
-    AvailableCoins(*locked_chain, vCoins, true, coinControl);
+
+    // Make sure we pass a CC
+    if (!coinControl) {
+        CCoinControl defaultCc;
+        coinControl = &defaultCc;
```

Comment:
This in invalid because the var is out of scope when you dereference it below, suspect that's behind the AddressSanitizer failure on CI.

That said, I would keep `coinControl` optional, as the default value of `0` for `nMinDepth` is equivalent to the absent / null condition for `coinControl`, it that it never fails the check. Would simplify diff overall.